common/src/main/cpp/classes/openxr_fb_spatial_entity.cpp

```diff
@@ -180,10 +184,47 @@ bool OpenXRFbSpatialEntity::is_tracked() const {
 	return OpenXRFbSpatialEntityExtensionWrapper::get_singleton()->is_entity_tracked(uuid);
 }
 
+Array OpenXRFbSpatialEntity::get_triangle_mesh() const {
```

Comment:
Shouldn't this return an `ArrayMesh` instead?

Reply:
No, this method is for if the developer wants the use the actual data for something. An `ArrayMesh` is immediately uploaded to the GPU, and if you want to access the data, Godot would actually have to pull it back down from the GPU again.

By returning an `Array` in the same format as `ArrayMesh` uses, it keeps the data on the CPU side, and allows the developer to inspect it, or manipulate it, possibly in order to construct their own `ArrayMesh` with it. (The demo does this - in order for the wireframe shader to work, we need to alter the mesh data.)

If the developer wanted to just use the unaltered mesh in rendering, then the idea is that they should call `create_mesh_instance()` which will give you a `MeshInstance3D` with an `ArrayMesh` on it.